### ISSUE TITLE
Signedness Checker emits false positive error with casted unsigned right shift expression

### ISSUE THREAD

**Issue by DmitriyShepelev** [typetools/checker-framework#3667]
The Signedness Checker emits a false positive error when the left operand of a casted (without an explicit signedness annotation) unsigned right shift expression is either a method parameter or field:
```
import org.checkerframework.checker.signedness.qual.Unsigned;
import org.checkerframework.checker.signedness.qual.Signed;

public class UnsignedRightShiftTestShort {
    void unsignedRightShiftWithParameter(int length) {
        byte b1 = (byte) (length >>> 24);
        byte b2 = (@Unsigned byte) (length >>> 24);
        byte b3 = (@Signed byte) (length >>> 24);
    }
}
```
Command: `javacheck -version -verbose -AprintAllQualifiers -processor SignednessChecker UnsignedRightShiftTestShort.java`

Output:
[UnsignedRightShiftTestShortOutput.txt](https://github.com/typetools/checker-framework/files/5241800/UnsignedRightShiftTestShortOutput.txt)

`javacheck` alias:
`alias javacheck='$CHECKERFRAMEWORK/checker/bin/javac'`

In particular, the line `byte b1 = (byte) (length >>> 24);` emits the false positive error. This line should be legal, according to [section 18.2.2 Permitted Shifts in the user manual](https://checkerframework.org/manual/#signedness-checker-permitted-shifts).
